File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/TickEFPData.java
package com.seagosoft.tws.data;

public class TickEFPData {
    public int tickerId;
    public int tickType;
    public double basisPoints;
    public String formattedBasisPoints;
    public double impliedFuture;
    public int holdDays;
    public String futureExpiry;
    public double dividendImpact;
    public double dividendsToExpiry;

    public TickEFPData(int tickerId, int tickType, double basisPoints, String formattedBasisPoints, double impliedFuture, int holdDays,
                       String futureExpiry, double dividendImpact, double dividendsToExpiry) {
        this.tickerId = tickerId;
        this.tickType = tickType;
        this.basisPoints = basisPoints;
        this.formattedBasisPoints = formattedBasisPoints;
        this.impliedFuture = impliedFuture;
        this.holdDays = holdDays;
        this.futureExpiry = futureExpiry;
        this.dividendImpact = dividendImpact;
        this.dividendsToExpiry = dividendsToExpiry;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/TickEFPData.kt
package com.seagosoft.tws.wrapper.structs

class TickEFPData(
    var tickerId: Int,
    var tickType: Int,
    var basisPoints: Double,
    var formattedBasisPoints: String,
    var impliedFuture: Double,
    var holdDays: Int,
    var futureExpiry: String,
    var dividendImpact: Double,
    var dividendsToExpiry: Double
)